src/payment/strategies/nonce/nonce-payment-strategy.ts

```diff
@@ -0,0 +1,54 @@
+import { isNonceLike } from '../..';
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+export default class NoncePaymentStrategy implements PaymentStrategy {
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator
+    ) { }
+
+    initialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    async execute(payload: OrderRequestBody, _options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        const { methodId, paymentData } = payment;
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        if (!paymentData || !isNonceLike(paymentData)) {
+            throw new MissingDataError(MissingDataErrorType.MissingPayment);
+        }
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, _options));
+
+        return this._store.dispatch(this._paymentActionCreator.submitPayment({
```

Comment:
would this be sent to bigpay? What will bigpay with `method_id: 'nonce'` ? 🤔

Reply:
when executing the strategy the method is is retrieved from the payload parameter, so if `method_id: 'bolt'` is sent in the payload  it will be submitted  to the respective gateway.

There is also a `gateway` parameter in the `OrderPaymentRequestBody` type, should I use that one instead? @capsula4

Reply:
I added the gateway parameter so it can be sent to the corresponding gateway.
But if the SDK is used by a third party application, would it  be posible to load the nonce strategy with a `nonce` methodId, but later use `bolt` as methodId when executing the strategy?

Reply:
I'm not exactly sure how this will be articulated in the background. It seems like we want to make a generic thing but in reality is not possible, as we need to pass `bolt`. I think we might be better off calling this `BoltPaymentStrategy`.

I don't have a thorough understanding of how BigPay works, maybe @davidchin can shed some light here.

Reply:
HI @capsula4 this is the tech design and diagram of this solution
https://intranet.bigcommerce.com/display/IDT/Bolt+-+Technical+Design

Reply:
Thanks for sharing this, so far, this doesn't seem to be a generic strategy.

Looking at this diagram, we might need a way to make "Bolt server" generic, that is: what URL should BC send the payment data to (along with the nonce), and then, what's the callback URL that bolt server will talk back to BC.

<img width="502" alt="Screen Shot 2020-05-19 at 10 35 55 am" src="https://user-images.githubusercontent.com/1621894/82271954-9b804e80-99bc-11ea-9683-aded7230c522.png">

It looks that's something BigPay would need to be able to handle, however I'm not the best person to talk about BigPay design. My understanding is that @davidchin  has already been part of some of the tech discussions around the Bolt integration, so I will defer this to him.